test_utils/utils.go

```diff
@@ -145,21 +155,30 @@ func CreateContainer(netconf string, k8sName string, ip string) (container_id, n
 	}
 
 	// Set up the env for running the CNI plugin
-	//TODO pass in the env properly
-	var k8s_env = ""
+	cni_env := []string{
+		"CNI_COMMAND=ADD",
+		"CNI_IFNAME=eth0",
+		"CNI_PATH=dist",
+		fmt.Sprintf("CNI_CONTAINERID=%s", container_id),
+		fmt.Sprintf("CNI_NETNS=%s", netnspath),
+	}
+
 	if k8sName != "" {
-		k8s_env = fmt.Sprintf("CNI_ARGS=\"K8S_POD_NAME=%s;K8S_POD_NAMESPACE=test;K8S_POD_INFRA_CONTAINER_ID=whatever\"", k8sName)
+		cniArgs := fmt.Sprintf("CNI_ARGS=K8S_POD_NAME=%s;K8S_POD_NAMESPACE=test;K8S_POD_INFRA_CONTAINER_ID=whatever", k8sName)
 
 		// Append IP=<ip> to CNI_ARGS only if it's not an empty string.
 		if ip != "" {
-			k8s_env = fmt.Sprintf("%s;IP=%s\"", strings.TrimRight(k8s_env, "\""), ip)
+			cniArgs = cniArgs + fmt.Sprintf(";IP=%s", ip)
 		}
+
+		cni_env = append(cni_env, cniArgs)
```

Comment:
@djosborne How come cni_env is an array of strings, but cniArgs is just a semicolon separated string? Does this even work appending them like this?